### PR TITLE
Fix android java exception parsing

### PR DESCRIPTION
**Pull request by kattrali**
## Goal

Ensure Android Java exceptions are formatted consistently with correct
stack traces and handledness

## Background

Some versions of Unity include a built-in thread uncaught exception
handler, which intercepts Java exceptions and emits them as log messages
if they occur on the main thread. The stacktrace string argument uses a
different format from Unity/C# exceptions, including the actual Java
exception name and message as the first line of the stacktrace.

## Changeset

This change detects android java exceptions by checking the error class
for "AndroidJavaException" then parsing the stacktrace according to the
java format. It fixes a case where the message / actual exception name
is placed into the message.

Added additional logic to process Java exceptions normally:

* Unwrap the "actual" error class from the log message
* Format the Java-style stack trace to match C# lines
* Set handled state for native Java exceptions to be unhandled

## Tests

* Added additional parsing tests for stacktrace formats

## Review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: src/BugsnagUnity/Payload/Exception.cs [hunk 1]
```diff
@@ -117,15 +103,31 @@ internal static Exception FromSystemException(System.Exception exception, System
...
 
-    internal static Exception FromUnityLogMessage(UnityLogMessage logMessage, System.Diagnostics.StackFrame[] stackFrames)
+    public static Exception FromUnityLogMessage(UnityLogMessage logMessage, System.Diagnostics.StackFrame[] stackFrames, Severity severity)
     {
-      var match = Regex.Match(logMessage.Condition, @"^(?<errorClass>\S+):\s*(?<message>.*)", RegexOptions.Singleline);
+      var match = Regex.Match(logMessage.Condition, ErrorClassMessagePattern, RegexOptions.Singleline);
 
       var lines = new StackTrace(logMessage.StackTrace).ToArray();
 
+      var handledState = HandledState.ForUnityLogMessage(severity);
```
[the remark this answers is not in the document]
Comment (kattrali) [reply src/BugsnagUnity/Payload/Exception.cs]: Ah, missed this comment. Yes, it could. It goes out of scope and is garbage collected right after in either case.
